diff --git a/es/hooks/useDelayState.js b/es/hooks/useDelayState.js
index 219d9545c4630f3061e00cf3f764abfedf6a2f74..73e43f949d7ee9cb440f816dbb035c6a94cd84e9 100644
--- a/es/hooks/useDelayState.js
+++ b/es/hooks/useDelayState.js
@@ -19,6 +19,7 @@ export default function useDelayState(defaultValue) {
       delayRef.current = null;
     }
   });
+  React.useEffect(() => cancelPending, [cancelPending]);
   const setDelayValue = useEvent((nextValue, immediatelyOrDelay) => {
     const delayConfig = immediatelyOrDelay || {
       frame: 1
diff --git a/lib/hooks/useDelayState.js b/lib/hooks/useDelayState.js
index e59989e3820a7cdf8f8197c1308e29fe09bdd3d6..2e61efabd1810bc50be0f3f131cc6640eeb87852 100644
--- a/lib/hooks/useDelayState.js
+++ b/lib/hooks/useDelayState.js
@@ -28,6 +28,7 @@ function useDelayState(defaultValue) {
       delayRef.current = null;
     }
   });
+  React.useEffect(() => cancelPending, [cancelPending]);
   const setDelayValue = (0, _useEvent.default)((nextValue, immediatelyOrDelay) => {
     const delayConfig = immediatelyOrDelay || {
       frame: 1
